scripts/retrieve_databundle_light.py

```diff
@@ -143,7 +143,7 @@ def download_and_unzip_zenodo(config, rootpath, hot_run=True, disable_progress=F
     """
     resource = config["category"]
     file_path = os.path.join(rootpath, "tempfile.zip")
-    destination = os.path.relpath(config["destination"])
+    destination = os.path.join(PREFIX, config["destination"])
```

Comment:
Thanks, @davide-f. That is the valid point. I have used BASE_DIR as the first try. And it worked fine as standalone for rule execution. But to make sure rules see each other and trigger, outputs and inputs needs to match. So `submodules/pypsa-earth/data/eez/eez_v11.gpkg` is not equal to absolute path `/home/yerbol/efuels/.../submodules/pypsa-earth/data/eez/eez_v11.gpkg` when it comes to input outputs of snakemake. 

With BASE_DIR instead `build_shapes` were not triggering `retrieve_databundle_light`. But `retrieve_databundle_light` worked just fine separately when calling it. CI test actually helped me to figure that out (https://github.com/pypsa-meets-earth/pypsa-earth/actions/runs/11314101704/job/31463655978)

BASE_DIR is good for handling files inside of the script, that are not tight to input/outputs of snakemake. If it is input/output, only PREFIX makes the full workflow intact and preserve rule connections.

Reply:
Do you mind talking on thursday about the prefix?
As PREFIX is taken out from the log inside the retrieve function, we do PREFIX = BASE_DIR, the result should be equivalent.

Moreover, don't have they the same meaning? I'd avoid a different naming unless necessary

Reply:
@davide-f, thanks, yes, we can discuss it in details in Thursday. In short, when importing the rules and defining prefix, all rule's inputs and outputs are appended with prefix, if they are explicitly set as input and output. 

I have performed several tests with `retrieve_databundle_light` and `build_shapes` rules using BASE_DIR and PREFIX. I have deleted `data/eez` which is used by `build_shapes`, then I run `build_shapes` rule. 
1) When using BASE_DIR inside `retrieve_databundle_light.py`, `build_shapes` were throwing error of missingInputException. Although `retrieve_databundle_light` rule can be executed separately just fine. Its outputs were absolute path. `build_shapes` could then be executed fine, but workflow is broken. Input of `build_shapes` is not exact to output of `retrieve_databundle_light`, although they are technically same file (one relative path PREFIX, and absolute path BASE_DIR).
2) When using PREFIX, outputs of `retrieve_databundle_light` perfectly matches with input of `build_shapes`. As a result, missing `data/eez` easily triggers `retrieve_databundle_light` rule when `build_shapes` are executed.

If having two names (BASE_DIR and PREFIX) is not preferred. We should have one: PREFIX. I will think about defining it in `_helpers.py` and replacing where BASE_DIR is used.

Reply:
Ok, but the content of variable PREFIX is the same as BASE_DIR in this PR now right?
How would the content of variable PREFIX change in the tests you are performing?
Do you have a branch somewhere where I can test and see what you are trying?

What operating system are you using for the tests? in windows there were known issues also on recognizing paths due to snakemake.

Reply:
> Ok, but the content of variable PREFIX is the same as BASE_DIR in this PR now right? How would the content of variable PREFIX change in the tests you are performing? Do you have a branch somewhere where I can test and see what you are trying?
> 
> What operating system are you using for the tests? in windows there were known issues also on recognizing paths due to snakemake.

The content of PREIX is `submodules/pypsa-earth`, while the content of BASE_DIR is `/home/user/.../some_project/submodule/pypsa-earth`. PREIX does not changes it is static relative path to pypsa-earth.

Regarding testing, I am testing on Windows and Linux in parallel, @GbotemiB was testing in Mac OS. The procedure and errors were consistent across all platforms. The issue of using BASE_FIR in `retrieve_databundle_light` had appeared in Linux and Windows as I tested.

To test the procedure, you can clone our `efuels-supply-potential` repository (https://github.com/open-energy-transition/efuels-supply-potentials) and checkout to `snakemake_workflow` branch. Ater submodule init and update of `submodule/pypsa-earth`, you will need to checkout to my branch `absolute_path_fixes`. You can also test with `BASE_DIR` by going back into commits.

Reply:
@davide-f, I am planning to get rid of using BASE_DIR in scripts, instead PREFIX will be used in each rule for consistency. The PREFIX will be defined `_helpers.py` and will be determined using BASE_DIR. What do you think?

Reply:
I've tested the repo and trying the procedure you suggested.
I still had issues with the workflow and I believe the reason is not really BASE_DIR/PREFIX. In this case we probably can always use BASE_DIR and it should be fine.

The problem lays into the function `datafiles_retrivedatabundle` that creates the vector of the files that retrieve_databundle shall produce.
The paths are by construction relative paths assuming the cwd is pypsa-earth, because they rely directly on the bundle_configs settings.

Probably the best solution lays in including in such function what you now call prefix.
I'd recommend however to find another name as prefix in pypsa-eur means another thing.
e.g. path_to_parent ? feel free to propose an alternative; it may also be a function.

Reply:
Thanks for the review, @davide-f. This "PREFIX" (path_to_parent) need to be added not only to `retrieve_databudle_light`, but other scripts, such as `build_shapes` and `build_demand_profiles`. I will look for `datafiles_retrivedatabundle` function to include path_to_parent.

Reply:
> I've tested the repo and trying the procedure you suggested. I still had issues with the workflow and I believe the reason is not really BASE_DIR/PREFIX. In this case we probably can always use BASE_DIR and it should be fine.
> 
> The problem lays into the function `datafiles_retrivedatabundle` that creates the vector of the files that retrieve_databundle shall produce. The paths are by construction relative paths assuming the cwd is pypsa-earth, because they rely directly on the bundle_configs settings.
> 
> Probably the best solution lays in including in such function what you now call prefix. I'd recommend however to find another name as prefix in pypsa-eur means another thing. e.g. path_to_parent ? feel free to propose an alternative; it may also be a function.

Can you please show what type of issue do you have? Have you tried only power model? If you can show the error message, then I can figure that out. Everything worked fine in my PCs (linux and windows) and Emmanuel's one.

Reply:
@davide-f, I have removed PREFIX and replaced with BASE_DIR everywhere. Also added BASE_DIR `build_renewable_profiles`. Then tested the full power workflow from subworkflow and pypsa-earth directory itself. Both worked as intended both in Linux and Windows.